src/yunohost/data_migrations/0021_migrate_to_bullseye.py

```diff
@@ -30,6 +30,60 @@
 N_NEXT_DEBAN = 11
 N_NEXT_YUNOHOST = 11
 
+VENV_BACKUP_SUFFIX = "_BACKUP_VENV"
+VENV_REQUIREMENTS_SUFFIX = "_req.txt"
+VENV_IGNORE = "ynh_migration_no_regen"
+
+def _get_all_venvs(dir,level=0,maxlevel=2):
+    # Using os functions instead of glob, because glob doesn't support hidden folders, and we need recursion with a fixed depth
+    result = []
+    for file in os.listdir(dir):
+        path = os.path.join(dir,file)
+        if os.path.isdir(path):
+            if os.path.isfile(os.path.join(path,VENV_IGNORE)):
+                continue
+            activatepath = os.path.join(path,"bin","activate")
+            if os.path.isfile(activatepath):
+                content = read_file(activatepath)
+                if "VIRTUAL_ENV" and "PYTHONHOME" in content:
+                    result.append(path)
+                    continue
+            if level<maxlevel:
+                result+=_get_all_venvs(path,level=level+1)
+    return result
+
+def _generate_requirements():
+
+    venvs = _get_all_venvs("/opt/")+_get_all_venvs("/var/www/")
+    for venv in venvs:
+        # Generate a requirements file from venv
+        os.system(f"bash -c 'source {venv}/bin/activate && pip freeze > {venv}{VENV_REQUIREMENTS_SUFFIX} && deactivate'")
+
+
+def _rebuild_venvs():
+
+    venvs = _get_all_venvs("/opt/")+_get_all_venvs("/var/www/")
+    for venv in venvs:
+        # Create a backup of the venv, in case there's a problem
+        if os.path.isdir(venv+VENV_BACKUP_SUFFIX):
+            rm(venv+VENV_BACKUP_SUFFIX, recursive=True)
+        backup = True
+        try:
+            cp(venv, venv+VENV_BACKUP_SUFFIX, recursive=True)
+        except:
```

Comment:
You probably should declare which errot you try catch

Reply:
Fixed in https://github.com/YunoHost/yunohost/pull/1479/commits/fc0266a62e8c1e9658b3829f15b3f5dbca03e06e